schema.ts

```diff
@@ -726,12 +743,54 @@ export function generateSchema(obj: ValidTypes): Schema {
   return ensureUnreachable(obj);
 }
 
+/**
+ * A wrapper for creating any schema definition.
+ *
+ * If you are creating a schema for an object (as opposed to a scalar or array),
+ * use the more specific {@link makeObjectSchema}.
+ *
+ * It is always recommended to use wrapper functions for creating top-level schema
+ * objects rather than specifying object literals. Wrappers validate your schemas
+ * at creation time, provide better TypeScript type inference, and can reduce
+ * boilerplate.
+ *
+ * At this time, this wrapper provides only better TypeScript type inference,
+ * but it may do validation in a future SDK version.
+ *
+ * @example
+ * ```
+ * coda.makeSchema({
+ *   type: coda.ValueType.Array,
+ *   items: {type: coda.ValueType.String},
+ * });
+ * ```
+ */
 export function makeSchema<T extends Schema>(schema: T): T {
   return schema;
 }
 
 export const PlaceholderIdentityPackId = -1;
 
+/**
+ * A wrapper for creating a schema definition for an object value.
```

Comment:
Other than the new ability to skip `type: coda.ValueType.Object`, does this provide any utility beyond `makeSchema`?

Reply:
It does some validation at import time, and because it's restricted to objects, the TypeScript inference is strong, TS will enforce that you're providing an object schema definition rather than just any valid schema definition. That would still work with `makeSchema` if you included `type: ValueType.Object`, but people seemed to find value in eliminating that boilerplate. But, I'm not opposed to consolidating these.